design/one-pager-package-format-v2.md

```diff
@@ -219,64 +211,89 @@ spec:
-such as an indexer. The `--context=unpack` flag will be passed when the package
-manager unpacks a package. The `--context` flag must only affect the annotations
-(`metadata.annotations`) of zero or more documents in the output stream. It may
-not alter the stream in any other way.
+Under this format:
+
+1. An on-disk package directory contains a `crossplane.yaml` file at its root.
+1. The `crossplane.yaml` file contains either a `Provider` or a `Configuration`.
+1. The package directory contains zero or more Crossplane specific custom
+   resources (e.g. a `Composition`). Crossplane is not opinionated about where
+   under the directory these files exist.
+
+The software responsible for building an OCI image from the on-disk format is
+outside the scope of this document, but is expected to:
+
+* Append the content of `crossplane.yaml` to `package.yaml`
+* Discover and append all Crossplane resources contained to `package.yaml`
+* Create a single-layer OCI image containing `package.yaml`.
+
+The aforementioned software may also be responsible for reading additional
+metadata from the package directory and including it in the OCI image.
+
+### Additional Metadata
+
+The Crossplane package format allows for two forms of additional metadata -
+passthrough metadata and supplmentary metadata.
+
+Passthrough metadata is opaque to the package manager and Crossplane in general,
+but is passed through from the package OCI image to the API server against which
+Crossplane is running. Examples of passthrough metadata could include:
+
+* Details about a provider or configuration, such as its maintainer or website.
+* Information about packaged resources (CRDs, XRDs, etc) that is opaque to
+  Crossplane, but that is interesting to other clients of the API server against
+  which Crossplane is running.
+
+Passthrough metadata must be included as annotations on the objects in the
+`/package.yaml` file. Annotations scoped to the entire package should be on the
+`Provider` or `Configuration` resource, and will be propagated unmodified by
+Crossplane to their associated `ProviderRevision` or `ConfigurationRevision`.
+Annotations scoped to a particular resource (e.g. a CRD) should be included on
+that resource. Additional metadata must not be attached to `/package.yaml` in
+any other way, including labels or additional resources.
+
+Metadata that is not consumed by Crossplane and that does not need to be passed
+through to the API server is known as supplementary metadata. This metadata may
+be included in any format its author and consumer(s) see fit, as long as it does
+not interfere with `/package.yaml`.
+
+Note that Crossplane is currently unopinionated about what additional metadata a
+package should contain. In future Crossplane may recommend standard additional
+metadata in order to ensure consumers of such metadata (e.g. package registries
+and user interfaces) use consistent metadata.
```

Comment:
@rbwsam @rathpc I feel like we could probably take the existing detail about package metadata from the v1 package manager design and the UI schema design and refactor it into something like `design-doc-recommended-additional-metadata.md`. What do you think?

Reply:
That makes sense to me. I also think some updates need to be made to https://github.com/crossplane/crossplane/blob/master/design/design-doc-upbound-ui-spec.md but that will be independent of that refactoring